### PR TITLE
2 - Klasyfikacja pasażerów Titanica

### PR DESCRIPTION
**Pull request by nuklearbanana**
Nazwa pliku: Titanic_1_and_2 
Zawiera się w nim również pierwsze zadanie. 
Zadanie drugie znajduje się na końcu.

### CODE REVIEW

**Review by M4jonez — COMMENTED** [review]
1.Do znalezienia najlepszych modeli używasz accuracy, a samo accuracy może dawać mylne wyniki przez niezbalansowanie klas (precision, recall albo F1-score może wykryłyby coś czego nie pokazuje accuracy)
2.Fajna macierz pomyłek na koniec, dużo pokazuje o faktycznej jakości modelu

**Review by fidok15 — APPROVED** [review]
- słabo czytelne wykresy 
- dużo opisów na + 
- data leakage najpierw powinieneś podzielić zbiór na treningowy i testowy a potem uzupełniać na podstawie treningowego
- niepoprawne usunięcie Fare które niosło informacje 
- do pokazania lepiej użyć [classification_report](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.classification_report.html) i [confusion_matrix](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.confusion_matrix.html)
- dobre podejście do przeskalowania danych z Age 
- nie użyłeś nigdzie final_data_without_age więc nie wiem po co rozdzielałeś 
- i jeszcze apropo data leakage najpierw dobierasz parametry modelu na zbiorze treningowym, a potem dopiero sprawdzasz jak model sobie radzi na zbiorze testowym

**Review by olabarycka — COMMENTED** [review]
Dobrze pokazujesz cały proces wyboru najlepszego modelu: od prostych bazowych porównań (DummyClassifier) przez test kilku klasyfikatorów, aż do optymalizacji hiperparametrów GridSearchCV. Plus za walidację krzyżową i analizę macierzy pomyłek.

**Comment by nuklearbanana** [comment]
[Titanic_1_and_2.ipynb](https://github.com/user-attachments/files/22222607/Titanic_1_and_2.ipynb)

Poprawione zadanie 

@Dominik-Galus  mógłbym Cię prosić o ponowne otwarcie PR żeby to wrzucić?

**Comment by Dominik-Galus** [comment]
@dekompot może wrzucić zadanie na pr'a z 4 zadaniem? Bo autor nie może mieć dwóch pull requestów na tego samego brancha, bo bym zamknął to z 5 zadaniem ale koordynator jeszcze nie ocenił

**Comment by nuklearbanana** [comment]
@Dominik-Galus Wrzuciłem poprawiony plik na PR z 4 zadaniem

**Comment by sonervous7** [comment]
#### REVIEW poprawki
1. Podział danych na początku na train test - o to chodziło super sprawa. Jedynie zabrakło parametru stratify, tak aby zachować balans po podziale.
2. Jak już zapadła decyzja o usuwani 'Fare', to dobrze, że po podziale.
3. Binary encoding do płci - o to chodziło :D
4. Świetnie, że użyłeś mediany do imputacji 'Age' i dobry argument, dlaczego nie średnia. Dodatkowo super potwierdzenie (output), że po imputacji brak NaN.
5. Pięknie, że fit na train i potem transform na test, nie ma data leakage. StandardScaler po imputacji medianą spoko wybór (można rozważyć ew. robust)
6. Dobre połącznie w całość i drop po zastosowaniu scalera.
7. Prawidłowo wydzeliłeś walidacyjny z treningowych, wcześniej odseparowując testowy do końcowej oceny.
8. Dobre dwa modele klasyfikacyjne, nie ma już KMeans. Świetnie, że jest wyjście od baseline dummy. Super, że są inne metryki.
9. Można najpierw odpalić na default parametrach przed gridsearch te dwa modele.
10. GridSearch z parametrami zastosowany poprawnie. Tu trzeba tylko wiedzieć, że GridSearch sam sobie zrobił z tego drugiego traina 4 foldy i na tej podstawie dobrał parametry.
11. Zastanowił bym się nad tym, jak można poradzić sobie z niezbalansowanym zbiorem. Na początku używając stratify przy podziałach, ale też czy accuracy jest odpowiednie do szukania najlepszego modelu w tym przypadku? Klasa dead jest dominująca, więc model przeiwdujący głównie 0, może mieć wysoką accuracy nawet jeśli słabo wykrywa klasę. Może być tak, że optymalizowane są parametry w ten sposób właśnie, że lepiej zgadują klasę 0.
12. Git inicjalizacja modeli. W porządku ostateczna ewaluacja na testowym. Dobre wnioski końcowe.

Bardzo dobra poprawa. Największy błąd z poprzedniej wersji, czyli data leakage, został wyeliminowany. Widać, że się przyłożyłeś, mega fajnie.
